hpx/runtime/serialization/detail/polymorphic_nonintrusive_factory.hpp

```diff
@@ -189,15 +193,47 @@ namespace hpx { namespace serialization { namespace detail
 #define HPX_SERIALIZATION_REGISTER_CLASS_NAME(Class, Name)                    \
     namespace hpx { namespace serialization { namespace detail {              \
         template <> HPX_ALWAYS_EXPORT                                         \
-        char const* get_serialization_name<Class>()                           \
+        struct get_serialization_name<Class>                                  \
         {                                                                     \
-            return Name;                                                      \
-        }                                                                     \
+            char const* operator()()                                          \
+            {                                                                 \
+                return Name;                                                  \
+            }                                                                 \
+        };                                                                    \
     }}}                                                                       \
     template hpx::serialization::detail::register_class<Class>                \
         hpx::serialization::detail::register_class<Class>::instance;          \
 /**/
+#define HPX_SERIALIZATION_REGISTER_CLASS_NAME_TEMPLATE(Parameters, Template, Name) \
+    namespace hpx { namespace serialization { namespace detail {              \
+        Parameters                                                            \
```

Comment:
Unfortunately, the `HPX_ALWAYS_EXPORT` has to be placed in between the 'struct' keyword and the type name. Visual Studio is picky about this (this is true for type definitions only, for functions it still expect it to be in front of everything).